src/assets/js/archive.js

```diff
@@ -122,66 +125,93 @@ function createTableCell(content, dataClass) {
+  const codeElements = document.querySelectorAll('code');
+  const filteredElements = Array.from(codeElements).filter(function (element) {
+    return Array.from(element.childNodes).some(function (node) {
+      return node.nodeType === Node.TEXT_NODE && node.textContent.includes(FILE_NAME_PREFIX);
+    });
   });
+  return filteredElements;
+}
 
-  $(textNode).replaceWith(`unzip ~/Downloads/${macOSArm64ArchiveFilename}`);
-});
 
-/*
-releases: A list of Flutter releases
-base_url: link for sdk download link such as storage.googleapis.com...
-os: macos, windows, or linux
-[optional] arch: Only specify if there's additional architecture, such as arm64
-*/
+/**
+ * Update the download button for the latest release.
+ * @constructor
+ * @param {Array} releases - A list of Flutter releases 
+ * @param {string} base_url - link for sdk download link such as storage.googleapis.com...
+ * @param {string} os - macos, windows, or linux
+ * @param {string} [arch=''] - Only specify if there's additional architecture, such as arm64
+ */
 function updateReleaseDownloadButton(releases, base_url, os, arch = '') {
   const archString = !arch.length ? '' : `-${arch}`;
 
   const release = releases[0];
   const linkSegments = release.archive.split('/');
   const archiveFilename = linkSegments[linkSegments.length - 1]; // Just the filename part of url
-  const downloadLink = $(`.download-latest-link-${os}${archString}`);
 
   if (os === 'macos' && arch === 'arm64') {
     macOSArm64ArchiveFilename = archiveFilename;
   }
 
-  downloadLink
-    .text(archiveFilename)
-    .attr('href', `${base_url}/${release.archive}`);
-
+  const downloadLink = document.querySelectorAll(`.download-latest-link-${os}${archString}`);
+  downloadLink.forEach((link) => {
+    link.textContent = archiveFilename;
+    link.setAttribute('href', `${base_url}/${release.archive}`);
+  })
 
   //Update download-filename placeholders:
-  $(`.download-latest-link-filename-${os}${archString}`).text(archiveFilename);
-  $('.download-latest-link-filename').text(archiveFilename);
+  const downloadLinkOs = document.querySelectorAll(`.download-latest-link-filename-${os}${archString}`);
+  downloadLinkOs.forEach(function(element) {
+    element.textContent = archiveFilename;
+  });
+  
+  const genericDownloadLink = document.querySelectorAll('.download-latest-link-filename');
+  genericDownloadLink.forEach(function(element) {
+    element.textContent = archiveFilename;
+  });
 
   // Update inlined filenames in <code> element text nodes:
-  const fileNamePrefix = 'flutter_';
-  const code = $(`code:contains("${fileNamePrefix}")`);
-  const textNode = $(code).contents().filter(function () {
-    return this.nodeType === 3 && this.textContent.includes(fileNamePrefix);
+  const filteredElements = filterCodeElements();
+  filteredElements.forEach(function (node) {
+    const newText = node.textContent.replace(new RegExp(`^(.*?)\\b${FILE_NAME_PREFIX}\\w+_v.*`), `$1${archiveFilename}`);
```

Comment:
This regex doesn't seem to be working to replace the filenames on the install pages. I think because using `textContent` now combines it all and the part to be replaced is on the second line. Adding the [multiline flag](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions#advanced_searching_with_flags) (`m`) should get it working.

```suggestion
    const newText = node.textContent.replace(new RegExp(`^(.*?)\\b${FILE_NAME_PREFIX}\\w+_v.*`, 'm'), `$1${archiveFilename}`);
```